tronweb2/src/components/Job/Job.js

```diff
@@ -0,0 +1,99 @@
+import React,
+{
+  useState,
+  useEffect,
+} from 'react';
+import {
+  useParams,
+} from 'react-router-dom';
+import JobScheduler from '../JobScheduler';
+import JobSettings from '../JobSettings';
+import ActionGraph from '../ActionGraph';
+import { getJobColor, fetchFromApi } from '../../utils/utils';
+
+function jobDisplay(job) {
+  return (
+    <div className="row">
+      <div className="col-md-5">
+        <h2>Details</h2>
+        <div>
+          <table className="table details">
+            <tbody>
+              <tr>
+                <td>Status</td>
+                <td className={`text-${getJobColor(job.status)}`}>{job.status}</td>
+              </tr>
+              <tr>
+                <td>Node pool</td>
+                <td>{job.node_pool.name}</td>
+              </tr>
+              <tr>
+                <td>Schedule</td>
+                <td><JobScheduler scheduler={job.scheduler} /></td>
+              </tr>
+              <tr>
+                <td>Settings</td>
+                <td>
+                  <JobSettings
+                    allowOverlap={job.allow_overlap}
+                    queueing={job.queueing}
+                    allNodes={job.all_nodes}
+                  />
+                </td>
+              </tr>
+              <tr>
+                <td>Last success</td>
+                <td>{job.last_success}</td>
+              </tr>
+              <tr>
+                <td>Next run</td>
+                <td>{job.next_run}</td>
+              </tr>
+            </tbody>
+          </table>
+        </div>
+      </div>
+      <div className="col-md-7">
+        <h2>Action Graph</h2>
+        <ActionGraph actionData={job.action_graph} height={400} width={450} />
+      </div>
+    </div>
+  );
+}
+
+function Job() {
+  const [job, setJobData] = useState(undefined);
+  const { jobId } = useParams();
+
+  useEffect(() => {
+    document.title = jobId;
+    fetchFromApi(`/api/jobs/${jobId}?include_action_graph=1`, setJobData);
```

Comment:
nit: it's probably not a real issue, but there is a race condition here: if the `Job` component is unmounted before this `fetch` resolves, the `setJobData` call will fail. To be totally safe, you'd want to [cancel the `fetchFromApi` promise](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html) in this effects [cleanup](https://reactjs.org/docs/hooks-effect.html#effects-with-cleanup).

Reply:
👍 The Tron API can be pretty slow at times, so might as well handle this case - updated. Tested by making the tron playground master sleep before responding to requests, and got a warning before and no warning after the change, when I clicked away from the job page before it loaded.